datadog/fwprovider/resource_datadog_restriction_policy.go

```diff
@@ -0,0 +1,240 @@
+	response.Diagnostics.Append(response.State.Set(ctx, &state)...)
+}
+
+func (r *RestrictionPolicyResource) Update(ctx context.Context, request resource.UpdateRequest, response *resource.UpdateResponse) {
+	var state RestrictionPolicyModel
+	response.Diagnostics.Append(request.Plan.Get(ctx, &state)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	resourceId := state.ResourceId.ValueString()
+	body, diags := r.buildRestrictionPolicyRequestBody(ctx, &state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	resp, _, err := r.Api.UpdateRestrictionPolicy(r.Auth, resourceId, *body)
+	if err != nil {
+		response.Diagnostics.Append(utils.FrameworkErrorDiag(err, "error retrieving RestrictionPolicy"))
+		return
+	}
+	if err := utils.CheckForUnparsed(resp); err != nil {
+		response.Diagnostics.AddError("response contains unparsedObject", err.Error())
+		return
+	}
+	r.updateState(ctx, &state, &resp)
+
+	// Save data into Terraform state
+	response.Diagnostics.Append(response.State.Set(ctx, &state)...)
+}
+
+func (r *RestrictionPolicyResource) Delete(ctx context.Context, request resource.DeleteRequest, response *resource.DeleteResponse) {
+	var state RestrictionPolicyModel
+	response.Diagnostics.Append(request.State.Get(ctx, &state)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	id := state.ID.ValueString()
+	httpResp, err := r.Api.DeleteRestrictionPolicy(r.Auth, id)
+	if err != nil {
+		if httpResp != nil && httpResp.StatusCode == 404 {
+			return
+		}
+		response.Diagnostics.Append(utils.FrameworkErrorDiag(err, "error deleting restriction_policy"))
+		return
+	}
+}
+
+func (r *RestrictionPolicyResource) updateState(ctx context.Context, state *RestrictionPolicyModel, resp *datadogV2.RestrictionPolicyResponse) {
+	state.ID = types.StringValue(resp.Data.GetId())
+	state.ResourceId = types.StringValue(resp.Data.GetId())
+	data := resp.GetData()
+	attributes := data.GetAttributes()
+
+	if bindings, ok := attributes.GetBindingsOk(); ok && len(*bindings) > 0 {
+		state.Bindings = []*BindingsModel{}
+		for _, bindingsDd := range *bindings {
+			bindingsTfItem := BindingsModel{}
+			if principals, ok := bindingsDd.GetPrincipalsOk(); ok && len(*principals) > 0 {
+				bindingsTfItem.Principals, _ = types.ListValueFrom(ctx, types.StringType, *principals)
+			}
+			if relation, ok := bindingsDd.GetRelationOk(); ok {
+				bindingsTfItem.Relation = types.StringValue(*relation)
+			}
+
+			state.Bindings = append(state.Bindings, &bindingsTfItem)
+		}
+	}
+}
+
+func (r *RestrictionPolicyResource) buildRestrictionPolicyRequestBody(ctx context.Context, state *RestrictionPolicyModel) (*datadogV2.RestrictionPolicyUpdateRequest, diag.Diagnostics) {
+	diags := diag.Diagnostics{}
+	attributes := datadogV2.NewRestrictionPolicyAttributesWithDefaults()
+
+	if state.Bindings != nil {
+		var bindings []datadogV2.RestrictionPolicyBinding
+		for _, bindingsTFItem := range state.Bindings {
```

Comment:
if `bindings` is not provided or is empty, what would be the expected behavior ?

Reply:
```400 Bad Request: {"errors":["Invalid request: update requests must contain at least one binding."]}```

Reply:
```400 Bad Request: {"errors":["Invalid request: principals list cannot be empty; found empty list for relation ''"]}```
^ if i do `bindings {}`

Reply:
If you want to delete the restriction policy, it'd have to be like this 
```terraform destroy -target='datadog_restriction_policy.foo' ```